client/src/Components/Layouts/Ordersmanagement/index.jsx

```diff
@@ -26,35 +44,304 @@ class OrdersManagement extends Component {
+      }
+    }
+  };
+
+  clearFields = async () => {
+    await this.setState({
+      date:"", status:"", name:"", filter:false
+    })
+  }
+
+  render() {
+    const { RangePicker } = DatePicker;
+    const dateFormat = "DD-MM-YYYY";
+    if (!this.state.error) {
+      return (
+        <div className="">
+          <div className="ordersManagement-bars-container ordersManagement-main-container">
+            <Header title={"إدارة الطلبات"} Icon={<Icon type="carry-out" />} />
+            <div className="ordersManagement_sub-container">
+              <div>
+                <Button
+                  className="ordersManagement_addOrder-button"
+                  type="primary"
+                >
+                  إضافة طلب
+                </Button>
+                <div className="ordersManagement_filters-container">
+                  <div className="ordersManagement_filters-container-timeFilter">
+                    <p
+                      style={{ textAlign: "right" }}
+                      className="ordersManagemet_timePicker-lable"
+                    >
+                      إختر الفترة
+                    </p>
+                    <RangePicker
+                      placeholder={["من", "إلى"]}
+                      format={dateFormat}
+                      onChange={e => this.filter("date", e)}
+                      value={this.state.date}
+                    />
+                  </div>
+                  <Input
+                    id="statusInput"
+                    onChange={e => this.filter("status", e.target.value)}
+                    className="ordersManagement_status-filter-input"
+                    placeholder="الفلترة حسب الحالة :"
+                    value={this.state.status}
+                  />
+                  <Input
+                    value={this.state.name}
+                    onChange={e => this.filter("name", e.target.value)}
+                    className="ordersManagement_status-filter-input"
+                    placeholder="الفلترة حسب اسم الكابتن :"
+                  />
+                  <Button
+                    className="ordersManagement_filter-button"
+                    type="primary"
+                    onClick={this.clearFields}
+                  >
+                    إفراغ الحقول
+                  </Button>
+                </div>
+                <TableComponent
+                  pageName="orders"
+                //   viewPopup={viewPopup}
+                //   editPopup={editPopup}
+                  deletePopup={deletePopup}
+                  editHtml="<input type='text' placeholder='name' id='name'>"
+                  columns={
+                    this.state.filter === true
+                      ? this.state.filteredOrders
+                      : this.state.orders
+                  }
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    } else if (this.state.error.response.status === 204) {
```

Comment:
no need for this if statement, the empty table no make any sense, but handle the loading data from the back end.